cpp/source/rocketmq/Producer.cpp

```diff
@@ -87,7 +87,8 @@ ProducerBuilder& ProducerBuilder::withConfiguration(Configuration configuration)
 }
 
 ProducerBuilder& ProducerBuilder::withTopics(const std::vector<std::string>& topics) {
-  impl_->topicsOfInterest(topics);
+  auto topics_copy = topics;
```

Comment:
I do not see how the proposed change is superior to the original. Any reason for this change?

Reply:
1. The input parameter of `ProducerBuilder::withTopics` is a const reference, while the input parameter of `topicsOfInterest` is not a constant reference. The original approach cannot be compiled.

2. The internal implementation of `topicsOfInterest` will use `std:: vector<>. swap()`, which will cause the internal memory space held by the incoming `topics` to be swapped out, which does not comply with the semantics of read-only input parameters for `withTopics`.

Reply:
> The input parameter of ProducerBuilder::withTopics is a const reference, while the input parameter of topicsOfInterest is not a constant reference. The original approach cannot be compiled.
No, this is not true.  `ProducerImpl::topicsOfInterest(std::vector<std::string> topics)` copy happens implicitly.

> 2. The internal implementation of topicsOfInterest will use std:: vector<>. swap(), which will cause the internal memory space held by the incoming topics to be swapped out, which does not comply with the semantics of read-only input parameters for withTopics.
No, it is using the implicit copy of the topics.

Reply:
@yulangz